db/document.go

```diff
@@ -811,3 +811,113 @@ func (doc *document) MarshalWithXattr() (data []byte, xdata []byte, err error) {
 
 	return data, xdata, nil
 }
+
+// Equal returns true if both of the given attachment metadata maps are considered the same.
+func (a AttachmentsMeta) Equal(b AttachmentsMeta) bool {
+	if len(a) != len(b) {
+		return false
+	} else if a == nil && b != nil {
+		return false
+	} else if b == nil && a != nil {
+		return false
+	}
+
+	// Find attachments in B that aren't in A
```

Comment:
I've just realized that since we're checking len(a) = len(b) above, and checking every attachment in 'a' below, we don't need this additional check.  Apologies for not catching this before making my previous comment.